### ISSUE TITLE
(aws-apigateway):  Usage of addCorsPreflight and addProxy can lead to CloudFormation drift when used with SpecRestApi

### ISSUE THREAD

**Issue by alulam** [aws/aws-cdk#24601]
### Describe the bug

Hello,

I've noticed a situation that can lead to CloudFormation drift when leveraging the addCorsPreflight and addProxy method with the [SpecRestApi](https://docs.aws.amazon.com/cdk/api/v1/docs/@aws-cdk_aws-apigateway.SpecRestApi.html) Construct. 

The exact steps are here:

- T0: User creates an API definition via [SpecRestApi](https://docs.aws.amazon.com/cdk/api/v1/docs/@aws-cdk_aws-apigateway.SpecRestApi.html) attribute [ApiDefinition](https://docs.aws.amazon.com/cdk/api/v1/docs/@aws-cdk_aws-apigateway.ApiDefinition.html)
- T1: User adds some resources via other CDK methods such as [addCorsPreflight](https://docs.aws.amazon.com/cdk/api/v1/docs/aws-apigateway-readme.html#cross-origin-resource-sharing-cors) and [addProxy](https://docs.aws.amazon.com/cdk/api/v1/docs/aws-apigateway-readme.html#proxy-routes)
- T2: User updates the Open API definition and redeploys stack
- T3: The API Gateway resources created at T1 disappear in API Gateway, but still exist in the template. Therefore causing CloudFormation Drift.

The documentation is also kind of confusing on this situation. In one area it says[ "All Resources and Methods need to be defined as part of the OpenAPI specification file, and cannot be added via the CDK." ](https://docs.aws.amazon.com/cdk/api/v1/docs/@aws-cdk_aws-apigateway.SpecRestApi.html)

In the [example](https://docs.aws.amazon.com/cdk/api/v1/docs/@aws-cdk_aws-apigateway.SpecRestApi.html#example) though, it does show new methods being added after also using the ApiDefinition attribute which conflicts with the earlier message. 

Also, [here]( https://docs.aws.amazon.com/cdk/api/v1/docs/aws-apigateway-readme.html#openapi-definition) it states that It is possible to use the addResource() API to define additional API Gateway Resources. This is true but users can get themselves in the above situation if they update the Open API definition in a future deployment.

### Expected Behavior

I would expect that the addProxy and addCorsPreflight API resources to remain even after I change the OpenApi definition of the SpecRestApi.

### Current Behavior

After I change the OpenApi definition and redeploy, the resources generated by the addCorsPreflight and addProxy methods are overriden in API Gateway but still exist in the template. Also when I check for CloudFormation drift, I see the stack has drifted.

### Reproduction Steps

```
   // Create regular pets api based on template
   const api = new ApiGateway.SpecRestApi(this, 'pests-api', {
      apiDefinition: ApiGateway.ApiDefinition.fromInline({
        openapi: '3.0.2',
        paths: {
          '/pets': {
            get: {
              'responses': {
                200: {
                  content: {
                    'application/json': {
                      schema: {
                        $ref: '#/components/schemas/Empty',
                      },
                    },
                  },
                },
              },
              'x-amazon-apigateway-integration': {
                responses: {
                  default: {
                    statusCode: '200',
                  },
                },
                requestTemplates: {
                  'application/json': '{"statusCode": 200}',
                },
                passthroughBehavior: 'when_no_match',
                type: 'mock',
              },
            },
          },
        },
        components: {
          schemas: {
            Empty: {
              title: 'Empty Schema',
              type: 'object',
            },
          },
        },
      })
    });

   const preflightOptionsTest = {
      allowCredentials: true,
      allowMethods: ['POST', 'GET'],
      allowOrigins: [
        'https://example.website',
      ],
   };
   
   // use addCorsPreflight
   api.root.addCorsPreflight(preflightOptionsTest);

   // use addProxy
   const corsProxyMethodTest = api.root.addProxy({
     anyMethod: false,
   });

   corsProxyMethodTest.addCorsPreflight(preflightOptionsTest);

```

Steps: 
1. Run the above code for a stack and deploy
2. Change the open API definition in some way
3. Redeploy
4. Check for CloudFormation drift and see that the proxy and OPTIONS method on root have disappeared 

### Possible Solution

As a suggested fix, I would recommend modifying the internal implementations of addCorsPreflight and addProxy to modify the OpenAPI document rather than creating the resources outside of the OpenApi document when the user is using SpecRestApi

I would really like to use addCorsPreflight and addProxy methods for my SpecRestApi given that they greatly simplify CORS integration steps. 

### CDK CLI Version

2.43.0 (build 487870a)

### Node.js Version

v18.13.0

### OS

macOS Monterey 12.6.3 

### Language

Typescript

**Comment by kirankumar15** [comment]
+1 I am also seeing the same issue on my end

**Comment by alulam** [comment]
Is there any way I can specify [mode](https://docs.aws.amazon.com/cdk/api/v1/docs/@aws-cdk_aws-apigateway.CfnRestApi.html#mode) as merge instead of overwrite for my SpecRestApi?

**Comment by pahud** [comment]
Yeah I think we should fix those examples to avoid confusion. When you create an API definition via [SpecRestApi](https://docs.aws.amazon.com/cdk/api/v1/docs/@aws-cdk_aws-apigateway.SpecRestApi.html) attribute [ApiDefinition](https://docs.aws.amazon.com/cdk/api/v1/docs/@aws-cdk_aws-apigateway.ApiDefinition.html), you are not supposed to update the returned rest API resource with other construct methods like `addResource` or `addMethod`.

**Comment by alulam** [comment]
I tried looking into [merge mode](https://docs.aws.amazon.com/cdk/api/v1/docs/@aws-cdk_aws-apigateway.CfnRestApi.html#mode) which keeps the addProxy and addCorsPreflight resources but unfortunately I don't think all users of my application would always want the behavior that is caused by merging the API definitions. 

Also I tried looking into [OpenAPI extensions](https://docs.aws.amazon.com/apigateway/latest/developerguide/api-gateway-swagger-extensions-cors-configuration.html) for CORS but it looks like those are only for HTTP APIs rather than REST APIs
